src/ml_performance_monitoring/monitor.py

```diff
@@ -245,44 +249,32 @@ def tuple_to_event(
 
     def prepare_events(
         self,
-        flat: pd.DataFrame,
-        y: pd.DataFrame,
+        inference_data: pd.DataFrame,
         metadata: Dict[str, Any],
-        inference_id_to_inference_metadata: Dict[str, Dict[str, str]],
+        inference_metadata: List[Dict[str, str]] = [],
         timestamp: Optional[int] = None,
         params: Optional[Dict[str, Any]] = None,
     ) -> Sequence[Event]:
         events: List[Event] = []
 
         request_id = self.get_request_id()
-        for t in flat.itertuples(index=False, name=None):
+        for index, t in zip(
+            range(len(inference_data)),
+            inference_data.itertuples(index=False, name=None),
+        ):
```

Comment:
`for index, t in enumerate(inference_data.itertuples(index=False, name=None)):` more pythonic